### PR TITLE
PSL polish

### PR DESCRIPTION
**Pull request by alicia-mbta**
Polish work for the Proposed Sales Location page:

- Minor tweaks (increase radius, rewording sentences).
- sort locations by distance
- stronger address validation to accept a string with different location formats (address or lat/lon).

### CODE REVIEW

File: apps/site/lib/site_web/templates/fare_transformation/proposed_sales_locations.html.eex [hunk 1]
```diff
@@ -45,7 +45,7 @@
     <div class="container" style="display: flex">
       <%= alert_icon() %>
       <div class="col-md-6">
-        <p><b>There are no proposed sales locations nearby. Please try to search again with a different location that's closer to areas the MBTA run.</b></p>
+        <p><b>The address you entered is more than 100 miles from the nearest proposed sales location. Please use an address that's closer to an MBTA service area.</b></p>
```
Comment (alicia-mbta) [inline apps/site/lib/site_web/templates/fare_transformation/proposed_sales_locations.html.eex:48]: This specific text might potentially change after Yinka and Jess agree on it.

**Review by hannahpurcell — CHANGES_REQUESTED** [review]
I like that the input box only uses addresses now!  It looks great, and the magnifying glass always works ✅ 

I've been using "Boston Landing" (a location in Brighton) as my test location.  The results in prod are correct, the first 3 being:  Boston Landing, Arthur St @ Guest St - Stop & Shop, and Brighton Ave @ Cambridge St.  I confirm with the map that these are the 3 closest PSLs to Boston Landing itself.  ✅ 

Dev-blue is showing a different 3 results, farther from the place of origin.  Not yet sure why this is happening  🤔 but is it related to the conversion of placename to coordinates, or vice versa?

**Comment by alicia-mbta** [comment]
> Dev-blue is showing a different 3 results, farther from the place of origin. Not yet sure why this is happening 🤔 but is it related to the conversion of placename to coordinates, or vice versa?

Good observation! I looked into it and the difference seems to be because of the increase from 10 to 100 miles 😞
I'm not sure why though, since I also added the option to sort by distance...I'll see if I can dig deeper.

**Comment by alicia-mbta** [comment]
UPDATE: I should sort, then slice 🤦‍♀️

**Review by thecristen — COMMENTED** [review]
I like that you moved `SiteWeb.Plugs.TransitNearMe.formatted_address` to `GoogleMaps.Geocode`... would it make sense to move `SiteWeb.FareController.calculate_position` and/or `SiteWeb.FareTransformationController.check_address` there as well?

**Comment by alicia-mbta** [comment]
> I like that you moved `SiteWeb.Plugs.TransitNearMe.formatted_address` to `GoogleMaps.Geocode`... would it make sense to move `SiteWeb.FareController.calculate_position` and/or `SiteWeb.FareTransformationController.check_address` there as well?

Yes! I didn't think of that; I'll move those as well.

File: apps/util/lib/position.ex [hunk 2]
```diff
@@ -6,7 +6,7 @@ defprotocol Util.Position do
   def longitude(item)
 end
 
-defimpl Util.Position, for: Map do
+defimpl Util.Position, for: [Map, Fares.ProposedLocations.Location] do
```
[the remark this answers is not in the document]
Comment (alicia-mbta) [reply apps/util/lib/position.ex:9]: Yes, makes sense to do it that way -- maybe I was just trying to save lines of code 😅

**Review by hannahpurcell — APPROVED** [review]
Looks great to me!

**Review by thecristen — APPROVED** [review]

**Pull request merged**